text/0000-macros-1.1.md

```diff
@@ -0,0 +1,473 @@
+    let input = input.to_source();
+
+    // use syntex_syntax from crates.io to parse `input` into an AST
+
+    // use this AST to generate an impl of the `Serialize` trait for the type in
+    // question
+
+    // convert that impl to a string
+
+    // parse back into a token stream
+    return TokenStream::from_source(&impl_source).unwrap()
+}
+```
+
+Next, crates will depend on this such as:
+
+```toml
+[dependencies]
+serde-macros = "0.9"
+serde = "0.9"
+```
+
+And finally use it as such:
+
+```rust
+#![rustc_macro_crate(serde_macros)]
+extern crate serde;
+
+#[derive(Serialize)]
+pub struct Foo {
+    a: usize,
+    #[serde(rename = "foo")]
+    b: String,
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* This is not an interface that anyone would wish to stabilize in a void, there
+  are a number of known drawbacks to the current macro system in terms of how
+  it architecturally fits into the compiler. Additionally, there's work underway
+  to solve all these problems with macros 2.0.
+
+  As mentioned before, however, the stable version of macros 2.0 is currently
+  quite far off, and the desire for features like custom derive are very real
+  today. The rationale behind this RFC is that the downsides are an acceptable
+  tradeoff from moving a significat portion of the nightly ecosystem onto stable
+  Rust.
+
+* This implementation is likely to be less performant than procedural macros
+  are today. Round tripping through strings isn't always a speedy operation,
+  especially for larger expantions. Strings, however, are a very small
+  implementation detail that's easy to see stabilized until the end of time.
+  Additionally, it's planned to extend the `TokenStream` API in the future to
+  allow more fine-grained transformations without having to round trip through
+  strings.
+
+* Users will still have an inferior experience to today's nightly macros
+  specifically with respect to compile times. The `syntex_syntax` crate takes
+  quite a few seconds to compile, and this would be required by any crate which
+  uses serde. To offset this, though, the `syntex_syntax` could be *massively*
+  stripped down as all it needs to do is parse struct declarations mostly. There
+  are likely many other various optimizations to compile time that can be
+  applied to ensure that it compiles quickly.
+
+# Alternatives
+[alternatives]: #alternatives
+
+* Wait for macros 2.0, but this likely comes with the high cost of postponing a
+  stable custom-derive experience on the time scale of years.
+
+* Don't stabilize `macro` as a new crate, but rather specify that
+  `#[rustc_macro_derive]` has a stable-ABI friendly signature. This does not
+  account, however, for the eventual planned introduction of the `macro` crate
+  and is significantly harder to write. The marginal benefit of being slightly
+  more flexible about how it's run likely isn't worth it.
+
+* The syntax for defining a macro will likely be different in the macros 2.0
```

Comment:
Oh I think @cgswords was under the impression that procedural macros 2.0 would be something like `pub macro foo(cx: &mut Context, TokenStream) -> TokenStream`, but we may be misinterpreting!